controllers/virtualmachine/volumebatch/volumebatch_controller.go

```diff
@@ -140,22 +139,22 @@ func AddToManager(ctx *pkgctx.ControllerManagerContext, mgr manager.Manager) err
 					"for CnsRegisterVolume: %w", err)
 
 		}
+	}
 
-		// Watch for changes for PersistentVolumeClaim, and enqueue
-		// VirtualMachine which is the owner of PersistentVolumeClaim.
-		if err := c.Watch(source.Kind(
-			mgr.GetCache(),
-			&corev1.PersistentVolumeClaim{},
-			handler.TypedEnqueueRequestForOwner[*corev1.PersistentVolumeClaim](
-				mgr.GetScheme(),
-				mgr.GetRESTMapper(),
-				&vmopv1.VirtualMachine{},
-			),
-		)); err != nil {
-			return fmt.Errorf(
-				"failed to start VirtualMachine watch "+
-					"for PersistentVolumeClaim: %w", err)
-		}
+	// Watch for changes for PersistentVolumeClaim, and enqueue
```

Comment:
I'd still keep this watch guarded by AllDisksArePVCs or VMSharedDisks or InstanceStorage